### ISSUE TITLE
Reserved keywords in version 3.1.1

### ISSUE THREAD

**Issue by Tenzer** [mhart/dynalite#120]
We are using dynalite for our test suite instead of calling the proper DynamoDB service, and we've seen our tests which are interacting with dynalite fail after the release of version 3.1.1.

I've constructed a minimal test case to reproduce the issue.

First, create the table like this:
```
[jeppe@reventon ~]$ aws dynamodb create-table \
>     --endpoint-url http://localhost:4567 \
>     --table-name my-table \
>     --attribute-definitions '[{"AttributeName": "user_id", "AttributeType": "N"}, {"AttributeName": "day", "AttributeType": "S"}]' \
>     --key-schema '[{"AttributeName": "user_id", "KeyType": "HASH"}, {"AttributeName": "day", "KeyType": "RANGE"}]' \
>     --provisioned-throughput '{"ReadCapacityUnits": 100, "WriteCapacityUnits": 100}'
{
    "TableDescription": {
        "AttributeDefinitions": [
            {
                "AttributeName": "user_id",
                "AttributeType": "N"
            },
            {
                "AttributeName": "day",
                "AttributeType": "S"
            }
        ],
        "TableName": "my-table",
        "KeySchema": [
            {
                "AttributeName": "user_id",
                "KeyType": "HASH"
            },
            {
                "AttributeName": "day",
                "KeyType": "RANGE"
            }
        ],
        "TableStatus": "CREATING",
        "CreationDateTime": 1575563327.596,
        "ProvisionedThroughput": {
            "NumberOfDecreasesToday": 0,
            "ReadCapacityUnits": 100,
            "WriteCapacityUnits": 100
        },
        "TableSizeBytes": 0,
        "ItemCount": 0,
        "TableArn": "arn:aws:dynamodb:us-east-1:000000000000:table/my-table",
        "TableId": "bf943054-4d7d-42e0-78d8-e83fb182"
    }
}
```

You don't need to load any data into the table.

If you query against dynalite the query succeeds like this:
```
[jeppe@reventon ~]$ aws dynamodb query \
>     --endpoint-url http://localhost:4567 \
>     --table-name my-table \
>     --key-condition-expression 'user_id = :uid AND #d > :date' \
>     --expression-attribute-names '{"#d": "day"}' \
>     --expression-attribute-values '{":uid": {"N": "123"}, ":date": {"S": "2019-12-01"}}'
{
    "Items": [],
    "Count": 0,
    "ScannedCount": 0,
    "ConsumedCapacity": null
}
```
while in dynalite 3.1.1 it instead fails like this:
```
[jeppe@reventon ~]$ aws dynamodb query \
>     --endpoint-url http://localhost:4567 \
>     --table-name my-table \
>     --key-condition-expression 'user_id = :uid AND #d > :date' \
>     --expression-attribute-names '{"#d": "day"}' \
>     --expression-attribute-values '{":uid": {"N": "123"}, ":date": {"S": "2019-12-01"}}'

An error occurred (ValidationException) when calling the Query operation: Invalid KeyConditionExpression: Attribute name is a reserved keyword; reserved keyword: day
```

Doing the query against the proper DynamoDB shows it behaves like dynalite 3.1.0, which leads me to believe this is a bug in dynalite.

**Comment by mhart** [comment]
Ugh, sorry about that – I refactored the query parser to fix another bug with it (hence the patch release) and I obviously introduced another edge case. WIll fix ASAP

**Comment by mhart** [comment]
Should be fixed in v3.0.2 – sorry for the breakage 😞

**Comment by mhart** [comment]
(Which has just been published, to clarify)

**Comment by Tenzer** [comment]
> Should be fixed in v3.0.2 – sorry for the breakage 😞

I assume you mean v3.1.2 😉 
Thank you for fixing this so fast!
